docs/advanced/encryption.md

```diff
@@ -12,9 +12,10 @@ Hookshot supports end-to-bridge encryption via [MSC3202](https://github.com/matr
 
 In order for hookshot to use encryption, it must be configured as follows:
 - The `experimentalEncryption.storagePath` setting must point to a directory that hookshot has permissions to write files into. If running with Docker, this path should be within a volume (for persistency).
+- If hookshot had ever been started with a Sled-based crypto store, set `experimentalEncryption.useLegacySledStore` to `true`. Otherwise, omit that setting (or set it to `false`) to use the SQLite-based crypto store.
```

Comment:
Part of me wonders if this should be a `storePath` with a `sqlite://file` format, like we do for our other bridges. We may end up doing psql formats too, and then you'll have `useLegacySledStore`, `useLegacySqliteStore` and `postgresUri` :D

Reply:
What makes this less straightforward is that the `storagePath` is not a path to a single file, but to a path to a _directory_ in which the bot-sdk saves crypto-related files for every appservice user.

At the time of writing, this includes the following files for each appservice user:
- a JSON file created by the bot-sdk for some basic info about the user (currently just the ID of their crypto-enabled device, which rooms they're in, and some crypto-relevant metadata for those rooms) 
- the SQLite (or Sled :slightly_smiling_face:) crypto store file created by the rust-sdk, for the meat of the user's crypto state

Notably, the crypto store files created by the rust-sdk are per-user, not per-appservice.

These files are organized in subdirectories within `storagePath`, one for each appservice user (as well as for a "default" user representing the appservice itself).

Given this, it seems like the bot-sdk (or downstream apps like Hookshot) can't use a single file / connection URL for the whole crypto store until both the bot-sdk & rust-sdk support per-appservice stores.

Even if the rust-sdk were to support Postgresql and manage all appservice users in the same database, the bot-sdk would still need some refactoring to not need those per-user JSON files it creates.

Reply:
In short: I'd rather incur the potential inconvenience of (yet another) migration path for when & if Postgresql is supported, instead of potentially making a wrong early guess as to how the rust-sdk & bot-sdk might work with a single file / database for crypto state.

Reply:
Ah, right I forgot we have *both* a JSON store and a SQLite store for each user. That's more of an issue. Okay, suggest when postgres comes along we treat it as a part of a larger refactor to reduce file usage.